n3fit/src/n3fit/scripts/evolven3fit_new.py

```diff
@@ -0,0 +1,376 @@
+    )
+    subparsers = parser.add_subparsers(title="actions", dest="actions")
+    evolvefit_parser = subparsers.add_parser(
+        "evolve",
+        help="Evolves the fitted PDFs. The q_grid starts at the Q0 given by the theory but the last point is q_fin and its number of points can be specified by q_points. If a path is given for the dump option, the eko will be dumped in that path after the computation. If a path is given for the load option, the eko to be used for the evolution will be loaded from that path. The two options are mutually exclusive.",
+    )
+    eko_parser = subparsers.add_parser(
+        "produce_eko",
+        help="Produce the eko for the specified theory. The q_grid starts at the Q0 given by the theory but the last point is q_fin and its number of points can be specified by q_points. The x_grid starts at x_grid_ini and ends at 1.0 and contains the provided number of points. The eko will be dumped in the provided path.",
+    )
+    evolvefit_parser.add_argument(
+        "configuration_folder",
+        help="Path to the folder containing the (pre-DGLAP) fit result",
+    )
+    parser.add_argument(
+        "-q",
+        "--q_fin",
+        type=float,
+        default=None,
+        help="Final q-value of the evolution",
+    )
+    parser.add_argument(
+        "-p",
+        "--q_points",
+        type=int,
+        default=None,
+        help="Number of q points for the evolution",
+    )
+    parser.add_argument(
+        "-n",
+        "--n_cores",
+        type=int,
+        default=1,
+        help="Number of cores to be used",
+    )
+    evolvefit_parser.add_argument(
+        "-l",
+        "--load",
+        type=pathlib.Path,
+        default=None,
+        help="Path of the EKO to be loaded",
+    )
+    evolvefit_parser.add_argument(
+        "-d",
+        "--dump",
+        type=pathlib.Path,
+        default=None,
+        help="Path where the EKO is dumped (optional)",
+    )
+    evolvefit_parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        help="Force the evolution to be done even if it has already been done",
+    )
+    eko_parser.add_argument(
+        "theoryID", type=int, help="ID of the theory used to produce the eko"
+    )
+    eko_parser.add_argument(
+        "dump",
+        type=pathlib.Path,
+        help="Path where the EKO is dumped",
+    )
+    eko_parser.add_argument(
+        "-i",
+        "--x_grid_ini",
+        default=None,
+        type=float,
+        help="Starting point of the x-grid",
+    )
+    eko_parser.add_argument(
+        "-p",
+        "--x_grid_points",
+        default=None,
+        type=int,
+        help="Number of points of the x-grid",
+    )
+    args = parser.parse_args()
+    op_card_info = {
+        "ev_op_max_order": 10,
+        "ev_op_iterations": 1,
+        "n_integration_cores": args.n_cores,
+        "backward_inversion": "expanded",
+    }
+    t_card_info = {}
+    if args.actions == "evolve":
```

Comment:
You can consistently define the function to use inside the individual definition of subparsers.

Maybe, you can have a look at an old project of mine:
https://github.com/AleCandido/slipy/blob/main/src/slides_cli/__init__.py

Essentially, you can make use of `parser.set_default(func=...)`, see e.g. the simple [`new` subparser](https://github.com/AleCandido/slipy/blob/7b493056f47cb4f22c0c1132ebdcb79535b36327/src/slides_cli/new.py#L12-L16)

Reply:
For this I believe that, since we have only two subparsers, for the moment we can even leave as it is. Having moved the two different definitions to external functions, I believe it is now clear enough.

Reply:
Ok fine, you have just two subcommands, fine to keep the `if`.

In any case, split in a separate function the content of the `elif`:
https://github.com/NNPDF/nnpdf/blob/c45413d48f59ee07f9116c610abca0f16b5af5b0/n3fit/src/n3fit/scripts/evolven3fit_new.py#L122-L146

And since you expect to never reach the `else` branch, add it raising an error (you never know, at least the error will tell us explicitly that something is going wrong there).